### PR TITLE
Rename component StyledOcticon to Octicon

### PR DESCRIPTION
**Pull request by jonrohan**
This PR updates the `StyledOcticon` component to the name `Octicon`. The reason I wanted to do this is to align this name with the rest of the system. We have `Primer::Beta::Octicon` [in PVC](https://primer.style/view-components/components/beta/octicon/) and the design guidelines [reference Octicons](https://primer.style/design/components/icon).

image

**So this could be shipped easier and not as a breaking change**, I tried to leave in an export at `./StyledOcticon` so that the [old import paths are backwards compatible](https://github.com/primer/react/pull/3148/files#diff-454e9c64d4f69cbb10ef71af79deb1a29374d51372ee2aa9023194dab70cd170). If we don't have something as a fallback we might need to rename about 300 icon imports when integrating in dotcom.

Closes https://github.slack.com/archives/C03M1LZ0UR3/p1677022255226319

### Screenshots

No visual change

### Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: src/StyledOcticon/StyledOcticon.tsx [hunk 1]
```diff
@@ -1,18 +1,5 @@
...
-function Octicon({icon: IconComponent, ...rest}: OcticonProps) {
-  return <IconComponent {...rest} />
-}
-
-const StyledOcticon = styled(Octicon)<SxProp>`
-  ${({color, sx: sxProp}) => sx({sx: {color, ...sxProp}})}
-`
-
-export type StyledOcticonProps = ComponentProps<typeof StyledOcticon>
 export default StyledOcticon
```
Comment (colebemis) [inline src/StyledOcticon/StyledOcticon.tsx:4]: Can we add a deprecation tag to StyledOcticon?

```suggestion
/** @deprecated Use Octicon instead */
export default StyledOcticon
```

Comment (colebemis) [inline src/StyledOcticon/StyledOcticon.tsx:4]: Can we add a deprecation tag to StyledOcticon?

```suggestion
/** @deprecated Use Octicon instead */
export default StyledOcticon
```

**Review by colebemis — APPROVED** [review]
🚀

**Review by colebemis — APPROVED** [review]
🚀

File: src/StyledOcticon/StyledOcticon.tsx [hunk 2]
```diff
@@ -1,18 +1,5 @@
-import {IconProps} from '@primer/octicons-react'
-import React from 'react'
-import styled from 'styled-components'
-import sx, {SxProp} from '../sx'
-import {ComponentProps} from '../utils/types'
+// Leaving an export here for backwards compatibility
```
Comment (siddharthkp) [inline src/StyledOcticon/StyledOcticon.tsx:1]: very nice!

**Review by siddharthkp — APPROVED** [review]
Love it, ship it!

**Pull request merged**